context_test.go

```diff
@@ -0,0 +1,31 @@
+package river
+
+import (
+	"context"
+	"testing"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClientFromContext(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	client := &Client[pgx.Tx]{}
+	ctx = withClient(ctx, client)
+
+	require.Equal(t, client, ClientFromContext[pgx.Tx](ctx))
+
+	result, err := ClientFromContextSafely[pgx.Tx](ctx)
+	require.NoError(t, err)
+	require.Equal(t, client, result)
+
+	require.Panics(t, func() {
```

Comment:
What do you think about switching to the use of `require.PanicsWithError` here? Just makes sure we for sure got the right panic and didn't trigger some other unexpected one by accident.

Reply:
Another handy `require` helper I didn't yet know about, thanks! Changed this test to assert about the specific error throughout, and also added a changelog entry.kj